spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/TaskParser.java

```diff
@@ -16,63 +16,344 @@
 
 package org.springframework.cloud.dataflow.core.dsl;
 
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.ANDAND;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.ARROW;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.CLOSE_PAREN;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.DOUBLE_MINUS;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.COLON;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.GT;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.IDENTIFIER;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.LITERAL_STRING;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.LT;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.OPEN_PAREN;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.OROR;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.SEMICOLON;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.STAR;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.springframework.util.Assert;
+
 /**
- * Parser for task DSL that generates {@link AppNode}.
+ * Parse a single or composed task specification.
  *
- * @author Michael Minella
- * @author Patrick Peralta
+ * @author Andy Clement
  */
 public class TaskParser extends AppParser {
 
+	public final static ArgumentNode[] NO_ARGUMENTS = new ArgumentNode[0];
+	
 	/**
-	 * Task name (may be {@code null}).
+	 * Task name.
 	 */
-	private final String name;
+	private final String taskName;
 
 	/**
-	 * Construct a {@code TaskParser} without supplying the task name up front.
-	 * The task name may be embedded in the definition; for example:
-	 * {@code myTask = filejdbc}
+	 * Task DSL text.
+	 */
+	private final String taskDefinition;
+	
+	/**
+	 * Controls usage mode for the parser. If true then apps can take arguments, if
+	 * false then app definitions are expected which take no arguments (currently).
+	 */
+	private final boolean inAppMode;
```

Comment:
actually... I believe non-appMode is for when this parser is used by the composed-task-runner, correct?